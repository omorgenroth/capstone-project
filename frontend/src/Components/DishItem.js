import { FaCheckCircle, FaRegCircle } from 'react-icons/fa'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

DishItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string.isRequired,
  isSelected: PropTypes.bool,
  onSelect: PropTypes.func,
}

export default function DishItem({ id, title, isSelected, onSelect }) {
  return (
    <ItemWrapper onClick={() => onSelect(id)}>
      <ContentWrapper>
        <Title> {title}</Title>
        <p> Ingredient1, Ingredient2, Ingredient3 </p>
      </ContentWrapper>
      <IconWrapper>
        {isSelected ? (
          <CheckedIcon data-testid="checked" checked={isSelected} />
        ) : (
          <UncheckedIcon data-testid="unchecked" checked={isSelected} />
        )}
      </IconWrapper>
    </ItemWrapper>
  )
}

const ItemWrapper = styled.div`
  background-color: ${(props) => (props.checked ? 'blue' : 'var(--c-white)')};
  box-shadow: 0, 3px, 6px, rgba(0, 0, 0, 0.5);
  border-radius: 15px;
  font-size: 0.5rem;
  padding: 4px 4px 4px 10px;
  display: grid;
  grid-template-columns: 3fr 1fr;
  width: 85%;
  height: 60px;
  position: relative;
`
const Title = styled.h2`
  font-size: 1.5em;
`

const ContentWrapper = styled.div`
  grid-column: 1/2;
`

const IconWrapper = styled.div`
  grid-column: 2/3;
`

const CheckedIcon = styled(FaCheckCircle)`
  color: var(--c-gray);
  transform: scale(2);
  position: absolute;
  top: 25px;
  right: 30px;
`
const UncheckedIcon = styled(FaRegCircle)`
  color: var(--c-gray);
  transform: scale(2);
  position: absolute;
  top: 25px;
  right: 30px;
`
