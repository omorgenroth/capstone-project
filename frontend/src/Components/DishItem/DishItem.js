import { FaCheckCircle, FaRegCircle } from 'react-icons/fa'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

DishItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  isSelected: PropTypes.bool,
  onSelect: PropTypes.func,
}

export default function DishItem({ id, title, isSelected, onClick }) {
  return (
    <ItemWrapper onClick={() => onClick(id)} selected={isSelected}>
      <ContentWrapper>
        <Title> {title}</Title>
      </ContentWrapper>
      <IconWrapper>
        {isSelected ? (
          <CheckedIcon data-testid="checked" />
        ) : (
          <UncheckedIcon data-testid="unchecked" />
        )}
      </IconWrapper>
    </ItemWrapper>
  )
}

const ItemWrapper = styled.div`
  background-color: ${(props) =>
    props.selected ? 'var(--c-green)' : 'var(--c-white)'};
  box-shadow: 0px 0px 6px 3px rgba(0, 0, 0, 0.03);
  color: ${(props) => (props.selected ? 'var(--c-gray)' : 'var(--c-gray)')};
  border-radius: 15px;
  border: 2px solid var(--c-green);
  font-size: 0.5rem;
  padding: 4px 4px 4px 10px;
  display: grid;
  grid-template-columns: 3fr 1fr;
  width: 100%;
  height: 60px;
  position: relative;
`
const Title = styled.p`
  font-size: 1rem;
`

const ContentWrapper = styled.div`
  grid-column: 1/2;
`

const IconWrapper = styled.div`
  grid-column: 2/3;
`

const CheckedIcon = styled(FaCheckCircle)`
  color: var(--c-gray);
  transform: scale(2.5);
  position: absolute;
  top: 25px;
  right: 30px;
`
const UncheckedIcon = styled(FaRegCircle)`
  color: var(--c-gray);
  transform: scale(2.5);
  position: absolute;
  top: 25px;
  right: 30px;
`
