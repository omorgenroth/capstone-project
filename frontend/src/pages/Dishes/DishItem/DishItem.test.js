import { render } from '@testing-library/react'
import DishItem from './DishItem'

describe('DishItem', () => {
  it('renders correctly', () => {
    const { container } = render(
      <DishItem isSelected={false} id={1} title="My Dish" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with the CheckedIcon if isSelected is true', () => {
    const { getByTestId } = render(
      <DishItem isSelected={true} id={1} title="My Dish" />
    )
    expect(getByTestId('checked')).toBeInTheDocument()
  })

  it('renders with the UnCheckedIcon if isSelected is false', () => {
    const { getByTestId } = render(
      <DishItem isSelected={false} id={1} title="My Dish" />
    )
    expect(getByTestId('unchecked')).toBeInTheDocument()
  })
})
