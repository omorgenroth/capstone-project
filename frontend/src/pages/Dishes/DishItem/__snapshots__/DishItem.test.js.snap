// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishItem renders correctly 1`] = `
<div
  class="css-8tkvjm"
>
  <div
    class="css-1t8btbg"
  >
    <p
      class="chakra-text css-0"
    >
      My Dish
    </p>
  </div>
  <div
    class="css-1ptvyx4"
  >
    <svg
      data-testid="unchecked"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
      />
    </svg>
  </div>
</div>
`;
